### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by estefaniaa**
- Add command line options to start service by checking existing batch job
- Add throwError helper where we log error, send error to slack and stop service from running further

### CODE REVIEW

File: env.sample [hunk 1]
```diff
@@ -1,6 +1,5 @@
 AZURE_ML_WEB_SERVICE_HOST_BATCH=
 AZURE_ML_WEB_SERVICE_PRIMARY=
-AZURE_SALT=
```
Comment (estefaniaa) [inline env.sample]: Not needed

File: service.js [hunk 2]
```diff
@@ -147,10 +149,7 @@ var checkBatchStatus = function(jobId, resultFilename) {
 		request.get(config.azure.hostBatch + '/' + jobId + '?api-version=2.0')
 		.set('Authorization', 'Bearer '+ config.azure.primaryKey)
 		.end(function(err, result) {
-			if (err) {
-				log.error(err);
-				return;
-			}
+			if (err) throwError(err);
```
Comment (chriserwin) [inline service.js]: it's probably not 100% required, but adding a `return` here would probably be a good idea.

File: service.js [hunk 3]
```diff
@@ -345,21 +340,28 @@ var startBatchProcess = function() {
...
 	// autoCallback
 	function(err, results) {
-		if (err) {
-			log.error(err);
-		}
-		else {
-			log.info('Batch job ' + results.batchJob + ' has started');
-			checkBatchStatus(results.batchJob, blobOutputFilename);
-		}
+		if (err) throwError(err);
```
Comment (chriserwin) [inline service.js]: `return` here too

File: service.js [hunk 4]
```diff
@@ -345,21 +340,28 @@ var startBatchProcess = function() {
...
+		log.info('Batch job ' + results.batchJob + ' has started, output file ' + blobOutputFilename);
+		checkBatchStatus(results.batchJob, blobOutputFilename);
 	});
 };
 
 
 if (runOnce) {
-	startBatchProcess();
+	if (job || output) {
+		if (!(job && output)) throwError('Must specify both job ID AND output file, cannot specify one without the other');
```
Comment (chriserwin) [inline service.js]: `return` here too

File: service.js (hunk 2 above)
Comment (estefaniaa) [inline service.js]: Whoops! That's what happens when you do a find/replace

**Comment by estefaniaa** [comment]
@chriserwin Made more improvements, mostly to error handling, and borrowed your Slack helper.  

I opted out of doing a dry-run since this process can't really run without affecting output, and many of the steps are dependent on a `jobId` and output file being created in Azure. If you think I might be missing something that would be valuable with a dry-run, let me know!

**Comment by chriserwin** [comment]
This looks great. I haven't run the service, but everything looks good and the linter isn't complaining about anything. I'm going to merge so we can test it out.

**Pull request merged**